app/views/reports/regions/_child_data_tables.html.erb

```diff
@@ -1,277 +1,404 @@
+          </h3>
+        </div>
+        <div class="table-responsive">
+          <table id="cumulative-registrations-table" class="table-compact">
+            <colgroup>
+              <col class="w-lg-120px">
+              <col class="table-divider">
+              <col class="table-divider w-lg-200px">
+            </colgroup>
+            <thead>
+              <tr class="sorts" data-sort-method="thead">
+                <th class="row-label sort-label sort-label-small ta-left">
+                  <%= region_type.capitalize %>
+                </th>
+                <th class="row-label sort-label sort-label-small ta-left" data-sort-method="number" data-sort-default>
+                  Patients registered in <%= @period.to_s %>
+                </th>
+                <th class="row-label sort-label sort-label-small ta-left" data-sort-method="number">
+                  Cumulative patients
+                </th>
+              </tr>
+            </thead>
+            <tbody>
+              <tr class="row-total" data-sort-method="none">
+                <td class="type-title">
+                  <%= @region.name %>
                 </td>
                 <td class="ta-left">
-                  <%= number_with_delimiter(result["registrations"].values.last || 0) %>
+                  <%= number_with_delimiter(@data.last_value(:registrations)) %>
                 </td>
                 <td class="ta-left">
-                  <%= number_with_delimiter(result["cumulative_registrations"].values.last) %>
+                  <%= number_with_delimiter(@data.last_value(:cumulative_registrations)) %>
                 </td>
               </tr>
-            <% end %>
-          </tbody>
-        </table>
-      </div>
-      <div class="mb-2 pt-12px">
-        <p class="us-none fs-14px c-transparent">
-          <strong>Denominator:</strong> <%= t("denominator_copy") %>
-        </p>
+              <% data.each do |result| %>
+                <% next if result.dig("registrations", @period).nil? %>
+                <tr>
+                  <td class="ta-left">
+                    <%= link_to(reports_region_path(result.region, report_scope: region_type))do %>
+                      <%= result.region.name %>
+                    <% end %>
+                  </td>
+                  <td class="ta-left">
+                    <%= number_with_delimiter(result["registrations"].values.last || 0) %>
+                  </td>
+                  <td class="ta-left">
+                    <%= number_with_delimiter(result["cumulative_registrations"].values.last) %>
+                  </td>
+                </tr>
+              <% end %>
+            </tbody>
+          </table>
+        </div>
+        <div class="mb-2 pt-12px">
+          <p class="us-none fs-14px c-transparent">
+            <strong>Denominator:</strong> <%= t("denominator_copy") %>
+          </p>
+        </div>
       </div>
     </div>
   </div>
-</div>
+<% end %>
```

Comment:
This view is getting really hard to review now - can you break it out into two partials, one for each state of the feature flag?